### ISSUE TITLE
[TBB] [feature] Any chance we get 2017 sometime soon?

### ISSUE THREAD

**Issue by SSE4** [conan-community/community#70]
_From @p-groarke on August 5, 2018 16:37_

The 4.4 version doesn't compile on MSVC 2017 since it uses `std::binary_function` in a header. This function has been [removed ](https://en.cppreference.com/w/cpp/utility/functional/binary_function) in c++17. Thank you.

https://github.com/01org/tbb/issues/13

_Copied from original issue: conan-community/conan-tbb#7_

**Comment by p-groarke** [comment]
@SSE4 alleluia queue the violins will test soon

**Comment by p-groarke** [comment]
@SSE4 Still getting `The code execution cannot proceed because tbb_debug.dll was not found.` Same in release.

**Comment by memsharded** [comment]
@p-groarke how are you consuming the package? Could you share your reusing code? Does it fail in release too? Still MSVC 2017?

Maybe you need to do some ``imports`` to get the .dll close to your binary for running?

**Comment by p-groarke** [comment]
@memsharded So the problem is, by default one consumes libraries as static. However TBB doesn't work as a static library (but it still compiles, tries to link without error). As such, you need to copy the tbb dlls manually from the package folder next to your executable for it to work.

Would it be possible to update the recipe so it acts like a dynamic library when compiled as a static lib? Or throw an error stating TBB doesn't support static linking?

I don't know about the `imports` feature though.

**Comment by p-groarke** [comment]
@memsharded Is there a way to setup a post-build copy step in cmake? Maybe I'll open a feature ticket then. That is what I would have expected. I'll link an example project that I am using tonight.

**Comment by memsharded** [comment]
Not sure what you mean. It doesn't need to be a post-build step, as they are shared libraries from the dependencies, they can be copied when they are installed, even before starting the current build of the consumer. So if they are declared as imports ``conan install`` will bring them to the user folder, and from there you can build and run your project.

**Comment by p-groarke** [comment]
OK I will experiment with imports. Sorry to waste your time.

**Comment by memsharded** [comment]
No problem, glad to help :)
Do not hesitate to ask any further question!

**Comment by SSE4** [comment]
@p-groarke should we consider this one done or do you still need some help or have any questions?

**Comment by p-groarke** [comment]
@SSE4 I still think this should work "out of the box", but you can close it.

**Comment by SSE4** [comment]
I am closing the issue for now, as new version is available. please open new issue(s) for more specific problems with TBB.
